presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGeneratorContext.java

```diff
@@ -275,7 +277,9 @@ private boolean hasOrderBy()
                 .filter(s -> !groupByColumns.contains(s.getKey())) // remove the group by columns from the query as Pinot barfs if the group by column is an expression
                 .map(s -> s.getValue().getDefinition())
                 .collect(Collectors.joining(", "));
-
+        if (expressions.isEmpty()) {
+            throw new PinotException(PINOT_QUERY_GENERATOR_FAILURE, Optional.empty(), "No valid expressions in " + this);
```

Comment:
`this` will provide the memory address (which is not necessary for an error message). What is needed is `PinotQueryGeneratorContext.class.getName()` I guess.